File: packages/shared-types/src/workspace.ts
export interface Workspace {
  id: string;
  name: string;
  mode: 'standalone' | 'hub';
  createdAt: string;
  updatedAt: string;
  encryptionKeyVersion: number;
}

export interface CreateWorkspaceRequest {
  name: string;
  mode: 'standalone' | 'hub';
}

File: apps/hub-service/src/models/workspace.rs
use serde::{Deserialize, Serialize};

#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Workspace {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub owner_id: Option<String>,
    pub settings: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateWorkspaceRequest {
    pub name: String,
    pub description: Option<String>,
}
